lib/fw.c

```diff
@@ -1223,6 +1225,25 @@ int switchtec_fw_read_fd(struct switchtec_dev *dev, int fd,
 	return read;
 }
 
+/**
+ * @brief Write the body for a Switchtec firmware image file
```

Comment:
Definitely need to describe the fact that this function is reading the firmware from the device and writing it to a file...